src/Symfony/Component/Serializer/Normalizer/UidNormalizer.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Normalizer;
+
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
+use Symfony\Component\Serializer\Exception\NotNormalizableValueException;
+use Symfony\Component\Uid\AbstractUid;
+use Symfony\Component\Uid\Ulid;
+use Symfony\Component\Uid\Uuid;
+
+class UidNormalizer implements NormalizerInterface, DenormalizerInterface, CacheableSupportsMethodInterface
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @throws InvalidArgumentException
+     */
+    public function normalize($object, string $format = null, array $context = [])
+    {
+        if (!$object instanceof AbstractUid) {
+            throw new InvalidArgumentException('The object must be an instance of "\Symfony\Component\Uid\AbstractUid".');
+        }
+
+        return (string) $object;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsNormalization($data, string $format = null)
+    {
+        return $data instanceof AbstractUid;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws NotNormalizableValueException
+     */
+    public function denormalize($data, string $type, string $format = null, array $context = [])
+    {
+        try {
+            $uid = Ulid::class === $type ? Ulid::fromString($data) : Uuid::fromString($data);
```

Comment:
it is possible that this code runs while the Uid component is not installed?
should we throw a LogicException if that's the case? (do we have similar cases in the component?)

Reply:
There's no similar case in the components, that's the first time we (de)normalize from/to an object defined in another component.

Indeed, this code will run is the method is explicitely called. I've updated it to first check if AbstractUid class exists and, if not, throw a logic exception.